lighthouse-cli/chrome-launcher.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const childProcess = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const mkdirp = require('mkdirp');
+const net = require('net');
+const rimraf = require('rimraf');
+const ask = require('./ask');
+const chromeFinder = require('./chrome-finder');
+
+const spawn = childProcess.spawn;
+const execSync = childProcess.execSync;
+
+module.exports = class Launcher {
+  constructor(opts) {
+    opts = opts || {};
+    // choose the first one (default)
+    this.head = defaults(opts.head, true);
+    this.pollInterval = 500;
+    this.chromeInstances = [];
+    this.prepared = false;
+  }
+
+  flags() {
+    const flags = [
+      '--remote-debugging-port=9222',
+      '--no-first-run',
+      `--user-data-dir=${this.TMP_PROFILE_DIR}`
+    ];
+
+    if (process.platform === 'linux') {
+      flags.push('--disable-setuid-sandbox');
+    }
+
+    return flags;
+  }
+
+  prepare() {
+    switch (process.platform) {
+      case 'darwin':
+        this.TMP_PROFILE_DIR = unixTmpDir();
+        break;
+
+      case 'linux':
```

Comment:
yeah... I liked that better too ... Was using two switch statements. Merged to one and forgot to change.